libs/events/import/api/src/lib/import/events-import.service.ts

```diff
@@ -0,0 +1,365 @@
+  const fallbackDescription = getFallbackDesc(source)
+  const parsedDescription = htmlToSlate(
+    longDescription || shortDescription || castInfo || fallbackDescription
+  )
+
+  // we need to add type: 'paragraph' because that's how it was done in WP in the past
+  parsedDescription[0] = {...parsedDescription[0], type: 'paragraph'} as any
+
+  const parsedEvent = {
+    id: XMLEvent['$'].originId,
+    modifiedAt: new Date(XMLEvent['$'].lastUpdate || ''),
+    name: XMLEvent.Title[0].replace(/(<([^>]+)>)/gi, ''),
+    description: parsedDescription,
+    status: EventStatus.Scheduled,
+    imageUrl: getImageUrl(XMLEvent),
+    externalSourceId: XMLEvent['$'].originId,
+    externalSourceName: source,
+    location: XMLEvent.Location[0]?.LocationAdress[0].replace(/(<([^>]+)>)/gi, '') || '',
+    startsAt: start ? new Date(start.trim()) : null,
+    endsAt: end ? new Date(end.trim()) : null
+  }
+
+  return parsedEvent
+}
+
+const fetchAndTransformImage = async (url: string): Promise<ArrayBufferUpload> => {
+  try {
+    const response = await axios.get(url, {responseType: 'arraybuffer'})
+    const {data, headers} = response
+
+    const filename = 'transformed-image.jpg'
+    const mimetype = headers['content-type']
+    const arrayBuffer = data
+
+    const arrayBufferUpload = {
+      filename,
+      mimetype,
+      arrayBuffer
+    }
+
+    return arrayBufferUpload
+  } catch (error) {
+    throw Error('Error fetching and transforming image')
+  }
+}
+
+@Injectable()
+export class EventsImportService {
+  constructor(
+    @Inject(CACHE_MANAGER) private cacheManager: Cache,
+    private prisma: PrismaClient,
+    private mediaAdapter: MediaAdapterService
+  ) {}
+
+  async importedEvents({filter, order, skip, take, sort}: ImportedEventsResolverParams) {
+    const importableEvents = Promise.all(
+      providers.map(provider =>
+        provider.importedEvents({
+          filter,
+          order,
+          skip,
+          take,
+          sort,
+          cacheManager: this.cacheManager,
+          source: provider.name
+        })
+      )
+    )
+
+    try {
+      const values = await importableEvents
+      // for now we have only one provider, to be extended in the future when needed
+      return values[0]
+    } catch (e) {
+      console.error(e)
+    }
+  }
+
+  async importedEvent(filter: SingleEventFilter) {
+    const {id, source} = filter
+    switch (source) {
+      case Providers.AgendaBasel: {
+        return new AgendaBasel().importedEvent({id, cacheManager: this.cacheManager})
+      }
+    }
+  }
+
+  async createEvent(filter: CreateEventArgs) {
+    const {id, source} = filter
+    switch (source) {
+      case Providers.AgendaBasel: {
+        return new AgendaBasel().createEvent({
+          id,
+          cacheManager: this.cacheManager,
+          prisma: this.prisma,
+          mediaAdapter: this.mediaAdapter
+        })
+      }
+    }
+  }
+}
+
+class AgendaBasel implements EventsProvider {
```

Comment:
cachemanager & prisma should be passed down via Nestjs providers and not in methods.
Essentially make AgendaBasel an injectable service